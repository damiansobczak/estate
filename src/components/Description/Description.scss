@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.description {
  &__subtitle {
    color: $color-text-secondary;
  }

  &__title {
    margin: 20px 0;
  }

  &__favorite {
    display: inline-flex;
    align-items: center;
    font-weight: 600;
    padding: 12px 18px;
    border-radius: 30px;
    background-color: $color-neutral-200;
    border: 0;
    outline: 0;
    font-size: 0.85rem;
    cursor: pointer;
    transition: all 0.1s ease-in;

    &:hover {
      background-color: $color-accent;
      color: #fff;
    }

    span {
      margin-right: 6px;
      font-size: 1rem;
    }
  }

  &__text {
    color: $color-text-secondary;
    line-height: 1.5;
    margin-top: 30px;
  }
}
