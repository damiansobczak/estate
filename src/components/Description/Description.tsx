import React, { MouseEvent } from "react";
import "./Description.scss";
import { DescriptionProps } from "./Interfaces/DescriptionProps";

class Description extends React.Component<DescriptionProps> {
  addToFavorite = (event: MouseEvent, id: any) => {
    event.preventDefault();
    if (!localStorage.getItem(id)) {
      localStorage.setItem(id, JSON.stringify(id));
    }
  }
  render() {
    return (
      <div className="description">
        <div className="description__subtitle">{this.props.subtitle}</div>
        <h2 className="description__title">{this.props.title}</h2>
        <button className="description__favorite" onClick={(event) => this.addToFavorite(event, this.props.id)}>
          <span className="icon-heart"></span> Dodaj do ulubionych
        </button>
        <div className="description__text">{this.props.text.replace(/<br\>|<p>|<\/p>|/g, '')}</div>
      </div>
    );
  }
}

export default Description;
