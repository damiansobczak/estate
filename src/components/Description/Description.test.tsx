import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Description from "./Description";

const details = {
    id: 24,
    title: "Osiedle Złote Łąki",
    text: "Przedstawiamy najnowszą ofertę biura nieruchomości.",
    cords: {
        x: "51.748378",
        y: "19.464890"
    },
    address: `Ostrów Wielkopolski, ul. Jasna 56`,
    rent: "1500",
    media: "600",
    deposit: "1500",
    price: "199.999PLN",
    category: "Sprzedaż",
    tags: [`Liczba pokoi 2`, `98m2`, `Liczba pięter 2`],
    assets: {
        lazienka: "Pysznic, pralka",
        pokoje: "3",
        kuchnia: "Kuchenka, czajnik",
        balkon: "15m2",
        parking: "",
        metraz: "55m2"
    },
    image: null,
    background: null,
    gallery: null
}

describe("<Description />", () => {
    it("renders component correctly", () => {
        const tree = renderer.create(<Description details={details} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("displays favorite button", () => {
        const wrapper = shallow(<Description details={details} />);
        expect(wrapper.html().search('Dodaj do ulubionych')).toBeTruthy();
    });
});