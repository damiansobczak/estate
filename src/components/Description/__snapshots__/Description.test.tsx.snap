// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description /> renders component correctly 1`] = `
<div
  className="description"
>
  <div
    className="description__subtitle"
  >
    Sprawdź szczegóły tej oferty
  </div>
  <h2
    className="description__title"
  >
    Osiedle Złote Łąki
  </h2>
  <button
    className="description__favorite "
    onClick={[Function]}
  >
    <span
      className="icon-heart"
    />
     Dodaj do ulubionych
  </button>
  <div
    className="description__text"
  >
    Przedstawiamy najnowszą ofertę biura nieruchomości.
  </div>
</div>
`;
